### ISSUE TITLE
SparseCategoricalAccuracy y_true and y_pred shape

### ISSUE THREAD

**Issue by ziruizhuang** [tensorflow/tensorflow#49252]
## URL(s) with the issue:
https://www.tensorflow.org/api_docs/python/tf/keras/metrics/SparseCategoricalAccuracy

## Description of issue (what needs changing):
It seems that the usage example conflicts with documentation. I suppose that the usage example is correct https://github.com/tensorflow/tensorflow/blob/a4dfb8d1a71385bd6d122e4f27f86dcebb96712d/tensorflow/python/keras/metrics.py#L3474-L3478

In usage example, `y_true` takes ground truth labels as input and `y_pred` takes logits/probabilities as input
```
m = tf.keras.metrics.SparseCategoricalAccuracy()
m.update_state([[2], [1]], [[0.1, 0.6, 0.3], [0.05, 0.95, 0]])
m.result().numpy()
```

In documentation, it states that:
`y_true` and `y_pred` should have the same shape.

y_true | Ground truth values. shape =&nbsp;[batch_size,&nbsp;d0,&nbsp;..&nbsp;dN].
-- | --
y_pred | The predicted values. shape =&nbsp;[batch_size,&nbsp;d0,&nbsp;..&nbsp;dN].

The documentation may need update.

**Comment by Saduf2019** [comment]
@ziruizhuang 
Can you please let us know what has to be updated please share more specific details].

**Comment by qlzh727** [comment]
Fixed in https://github.com/keras-team/keras/commit/4244fba008d686f644b2ea3711e1897edf32432b.
